torchtune/models/qwen2/_tokenizer.py

```diff
@@ -345,6 +343,9 @@ def tokenize_messages(
         Raises:
             RuntimeError: If a message contains non-text content
         """
+        assert not isinstance(
+            self.prompt_template, ChatMLTemplate
+        ), "Using ChatMLTemplate with tokenize_messages will result in multiple <|im_*|> tokens wrapping each message."
```

Comment:
nit: we should add an actionable message as well, "Please use a different template or set to None"